### PR TITLE
Migration of account_journal_report_xls to 8.0

### PR DESCRIPTION
**Pull request by guewen**
- Corrected Pylint Warnings
- Added tests
- Added translation template
- Fixed a Many2many bug: https://github.com/odoo/odoo/issues/3751

### CODE REVIEW

**Comment by sbidoul** [comment]
:smile:

**Comment by guewen** [comment]
> :smile:

And I checked before ;-)

**Comment by guewen** [comment]
The failure is due to pylint warnings in account_financial_report_webkit, not related to my migration.

File: account_journal_report_xls/report/nov_account_journal_xls.py [hunk 1]
```diff
@@ -235,7 +235,7 @@ def __init__(self, name, table, rml=False, parser=False, header=True,
                                self.aml_cell_style_decimal]},
         }
 
-    def _journal_title(self, o, ws, _p, row_pos, xlwt, _xs):
+    def _journal_title(self, o, ws, _p, row_pos, _xs):
```
Comment (foutoucour) [inline account_journal_report_xls/report/nov_account_journal_xls.py:238]: need more clear arguments or docstrings

File: account_journal_report_xls/report/nov_account_journal_xls.py [hunk 2]
```diff
@@ -251,7 +251,7 @@ def _journal_title(self, o, ws, _p, row_pos, xlwt, _xs):
             ws, row_pos, row_data, row_style=cell_style)
         return row_pos + 1
 
-    def _journal_lines(self, o, ws, _p, row_pos, xlwt, _xs):
+    def _journal_lines(self, o, ws, _p, row_pos, _xs):
```
Comment (foutoucour) [inline account_journal_report_xls/report/nov_account_journal_xls.py:254]: need more clear arguments or docstrings

File: account_journal_report_xls/report/nov_account_journal_xls.py [hunk 3]
```diff
@@ -305,7 +305,7 @@ def _journal_lines(self, o, ws, _p, row_pos, xlwt, _xs):
             ws, row_pos, row_data, row_style=self.rt_cell_style_right)
         return row_pos + 1
 
-    def _journal_vat_summary(self, o, ws, _p, row_pos, xlwt, _xs):
+    def _journal_vat_summary(self, o, ws, _p, row_pos, _xs):
```
Comment (foutoucour) [inline account_journal_report_xls/report/nov_account_journal_xls.py:308]: need more clear arguments or docstrings

File: account_journal_report_xls/report/nov_account_journal_xls.py (hunk 1 above)
Comment (guewen) [inline account_journal_report_xls/report/nov_account_journal_xls.py:238]: I totally agree but I am not the author of the module and this PR is only a migration to v8, not a refactoring. IMO this point should have been raised and adressed before the inclusion of this module. (same remark for the other comments below)

**Comment by foutoucour** [comment]
Understood
:+1:

**Comment by sbidoul** [comment]
@guewen can you rebase?

**Comment by guewen** [comment]
Rebased

**Comment by yvaucher** [comment]
:+1:

**Comment by JehoG** [comment]
What's the status on this PR?

**Comment by bwrsandman** [comment]
:+1:

**Pull request merged**